.github/workflows/release.yml

```diff
@@ -0,0 +1,194 @@
+name: release
+on:
+  release:
+    types: [published]
+    tags:
+      - v*
+  # Uncomment this if you'd like to test the build+release process on PRs too
+  # (it will only test the build, it wont push to PyPI until a new release is tagged)
+  # push:
+  #   branches:
+  #     - master
+  # pull_request:
+  #   branches:
+  #     - master
+
+env:
+  LIBZIM_RELEASE: libzim_linux-x86_64-6.1.1
+  LIBZIM_LIBRARY_PATH: lib/x86_64-linux-gnu/libzim.so.6.1.1
+  LIBZIM_INCLUDE_PATH: include/zim
+  CYTHON_VERSION: 0.29.6
+
+jobs:
+  release:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        # TODO: expand this to cross-platform builds (see V2 approach below)
+        # os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: x64
+
+      - name: Cache libzim dylib
+        uses: actions/cache@master
+        id: cache-libzim
+        with:
+          path: libzim_linux
+          key: $LIBZIM_RELEASE-libzim-cache
+
+      - name:  Install libzim dylib & headers in system
+        if: steps.cache-libzim.outputs.cache-hit != 'true'
+        run: |
+          wget -q https://download.openzim.org/release/libzim/$LIBZIM_RELEASE.tar.gz
+          tar --gunzip --extract --file=$LIBZIM_RELEASE.tar.gz
+          mv $LIBZIM_RELEASE libzim_linux
+
+      - name: Sync libzim dylib & headers in system
+        run: |
+          sudo rsync -a $GITHUB_WORKSPACE/libzim_linux/$LIBZIM_LIBRARY_PATH /usr/lib/libzim.so
+          sudo rsync -a $GITHUB_WORKSPACE/libzim_linux/$LIBZIM_INCLUDE_PATH/ /usr/include/zim/
+
+      - name: Copy libzim release for building
+        run: |
+          rsync -a $GITHUB_WORKSPACE/libzim_linux/ $GITHUB_WORKSPACE/$LIBZIM_RELEASE/
+
+      - name: Build cython, sdist, and bdist_wheels
+        run: |
+          pip install --upgrade cython==$CYTHON_VERSION setuptools pip cibuildwheel
+          python3 setup.py build_ext --inplace
+          python3 setup.py sdist bdist_wheel
+          python -m cibuildwheel --output-dir wheelhouse
+
+      - uses: actions/upload-artifact@v1
+        with:
+          name: wheels
+          path: ./wheelhouse
+
+      - name: Push release to PyPI
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

Comment:
It seems that it will publish on PyPI on all push on master. Not only when we tag a release.

Reply:
This section at the top combined with this section should limit PyPI releases to only push tags on master starting with `v*`:
```python3
on:
  release:
    types: [published]
    tags:
      - v*
```

Reply:
Yes, but in the same time you write just after :
```
# Uncomment this if you'd like to test the build+release process on PRs too
  # (it will only test the build, it wont push to PyPI until a new release is tagged)
  # push:
  #   branches:
  #     - master
  # pull_request:
  #   branches:
  #     - master
```
If I uncomment it, ci will push the "non release" on PyPI.

Reply:
Yes, that's intentional as it's useful for releases to test PyPI `https://test.pypi.org/legacy/`, where you can push a release on every PR and push in order to make sure the build/release process works correctly.  If you want me to remove it I can, the alternative is to just test locally with test.pypi.org and not use Github Actions for test releases.

Reply:
(removed)